db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  # Dummy classes, to make migration possible across version changes
+  class Account < ApplicationRecord
+    has_one :user, inverse_of: :account
+    has_one :notification_policy, inverse_of: :account
+  end
+
+  class User < ApplicationRecord
+    belongs_to :account
+  end
+
+  class NotificationPolicy < ApplicationRecord
+    belongs_to :account
+  end
+
+  def up
+    User.includes(account: :notification_policy).find_each do |user|
+      deserialized_settings = Oj.load(user.attributes_before_type_cast['settings'])
+      policy = user.account.notification_policy || user.account.build_notification_policy
```

Comment:
Are there scenarios where any accounts will already have a policy? Assuming these migrations are run close together, it seems unlikely ... curious if I'm missing something obvious or if we're just being defensive here for the edge case where migrations are run w/out downtime and an account could hypothetically create a policy between the migrations running...

Reply:
Yes, this is programmed defensively because there are situations when admins forget to run migrations.